pinot-common/src/main/java/org/apache/pinot/common/assignment/InstancePartitions.java

```diff
@@ -58,25 +60,37 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class InstancePartitions {
   private static final char PARTITION_REPLICA_GROUP_SEPARATOR = '_';
+  private static final String PARTITIONS_KEY = "partitions";
+  private static final String INSTANCE_SEPARATOR = "/";
 
   private final String _instancePartitionsName;
-  private final Map<String, List<String>> _partitionToInstancesMap;
+  private final Map<String, List<String>> _partitionWithReplicaGroupToInstancesMap;
+  private final Map<Integer, List<String>> _partitionToInstancesMap;
   private int _numPartitions;
   private int _numReplicaGroups;
 
   public InstancePartitions(String instancePartitionsName) {
     _instancePartitionsName = instancePartitionsName;
+    _partitionWithReplicaGroupToInstancesMap = new TreeMap<>();
     _partitionToInstancesMap = new TreeMap<>();
   }
 
   @JsonCreator
   private InstancePartitions(
       @JsonProperty(value = "instancePartitionsName", required = true) String instancePartitionsName,
-      @JsonProperty(value = "partitionToInstancesMap", required = true)
-          Map<String, List<String>> partitionToInstancesMap) {
+      @JsonProperty(value = "partitionWithReplicaGroupToInstancesMap", required = true)
+          Map<String, List<String>> partitionWithReplicaGroupToInstancesMap,
+      @JsonProperty(value = "partitionToInstancesMap")
```

Comment:
To get a better understanding of the new information in InstancePartitions, can you update the PR description with an example of what it looks like now ?

Reply:
An example is added to the description of the current PR.